qiskit/circuit/parameterexpression.py

```diff
@@ -90,6 +95,36 @@ def bind(self, parameter_values):
 
         return ParameterExpression(free_parameter_symbols, bound_symbol_expr)
 
+    def resolve(self, parameter_values):
+        """Returns a value with replacement Parameters.
+
+        Args:
+            parameter_values (dict): Mapping from Parameters in self to the
+                                     ParameterExpression instances with which they
+                                     should be replaced. All of the parameters
+                                     in this expression must be included.
+        Raises:
+            CircuitError:
+                - If parameter_values contains Parameters outside those in self.
+                - If parameter_values does not contain all the Parameters in self.
+                - If parameter_values contains non-real or non-numeric values.
+
+        Returns:
+            float: a value with the specified parameters replaced.
+        """
+        self._raise_if_passed_unknown_or_missing_parameters(parameter_values.keys())
+        self._raise_if_passed_non_real_value(parameter_values)
+
+        keys = sorted(parameter_values.keys())
```

Comment:
Instead of adding `__lt__`, etc. to `Parameter`s, can we use a `key` func in `sorted` (maybe by `Parameter.name`)?